scripts/globals/weaponskills.lua

```diff
@@ -1237,6 +1244,32 @@ xi.weaponskills.handleParry = function(attacker, target, missChance, guaranteedH
         if target:getSystem() == xi.eco.BEASTMEN or target:isPC() then
             missChance = 1
         end
+
+        if target:isPC() then
+            target:trySkillUp(xi.skill.PARRY, target:getMainLvl())
+        end
+    end
+
+    return missChance
+end
+
+xi.weaponskills.handleGuard = function(attacker, target, missChance, guaranteedHit)
+    local gHit = guaranteedHit or false
+    if
+        target:getMainJob() == xi.job.MNK or
+        target:getMainJob() == xi.job.PUP
+    then
+        if
+            target:getGuardRate(attacker) > math.random(100) and
+            target:isFacing(attacker) and
+            target:isPC() and
+            not gHit
+        then
```

Comment:
Should we also check if the attack's level is high enough to level the player's guard skill? With this if statement alone, if I'm reading this correctly, they should be able to skill up from lvl 1 mobs at 75.
Or is this being handled in trySkillUp?

Reply:
good question, ill double check

Reply:
Good catch - I was using the defender, not the attacker in the scenario.
Updated all calls added for trySkillUp to pass in the attacker's level.

Reply:
See line 1271
Prev rev had defender:getMainLvl()
trySkillUp uses that param to determine skill up chances